11_notify/n0npax/pkg/middleware/slowerer.go

```diff
@@ -0,0 +1,22 @@
+package middleware
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+)
+
+// SlowRequest introduces delay n miliseconds
+func SlowRequest(duration string) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		if minDuration, err := time.ParseDuration(duration); err != nil {
```

Comment:
I think it would have been better to do this validation when you create the middleware and crash the app if it fails, that way you do the check once on startup, and if it passes, you know every time the endpoint is hit, it will always work.
```go
func SlowRequest(duration time.Duration) gin.HandlerFunc {
        if duration < 0 {
                duration = 0 // You could also choose to throw an error here
        }
	return func(c *gin.Context) {
                ...
        }
}
```

This requires a change to the method signature but I believe its worth it.

Reply:
fixed.
I skipped `if duration < 0` and added comment that negative duration is treated like a 0.